### PR TITLE
Adding file extensions to gitattributes

### PR DESCRIPTION
**Pull request by Rafi993**
## What type of PR is this?

- [ ] Refactor
- [ ] Feature
- [ ] Bug Fix
- [x] Optimization
- [ ] Documentation Update

## Description

Adding file extensions to .gitattributes for better diff. It is mostly inspired from https://tekin.co.uk/2020/10/better-git-diff-output-for-ruby-python-elixir-and-more

## Related Tickets & Documents
Closes https://github.com/forem/forem/issues/10946

## QA Instructions, Screenshots, Recordings

Try doing git diff using git cli before and after this commit on a ruby file

## Added tests?

- [ ] yes
- [x] no, because they aren't needed
- [ ] no, because I need help

## Added to documentation?

- [ ] docs.forem.com
- [ ] readme
- [x] no documentation needed

### CODE REVIEW

**Review by citizen428 — APPROVED** [review]

File: .gitattributes [hunk 1]
```diff
@@ -3,3 +3,7 @@
 # means that when checking out code, LF endings get converted
 # to CRLF.
 # *.svg text eol=lf
+*.rb   diff=ruby
+*.rake diff=ruby
+*.css  diff=css
+*.html diff=html
```
Comment (citizen428) [inline .gitattributes:9]: Do you want to add HTML diffs for ERb files too?

**Review by citizen428 — APPROVED** [review]

File: .gitattributes (hunk 1 above)
Comment (Rafi993) [reply .gitattributes:9]: I thought so too but isn't erb little different (since it can have ruby code inside it)

Comment (citizen428) [reply .gitattributes:9]: Yeah, but a lot of editors work quite ok if you set ERb's file type to HTML, so I guess it's worth at try.

Comment (Rafi993) [reply .gitattributes:9]: I have added `*.erb  diff=html` now. Thank you @citizen428

File: .gitattributes [hunk 2]
```diff
@@ -3,3 +3,8 @@
...
+*.erb  diff=html
```
Comment (rhymes) [inline .gitattributes:10]: there's a missing newline at the end of the file

**Review by rhymes — COMMENTED** [review]
Hi @Rafi993, thanks for the change.

What about frontend files? JS, JSX and so on? Should we add those as well

File: .gitattributes (hunk 2 above)
Comment (Rafi993) [reply .gitattributes:10]: > Hi @Rafi993, thanks for the change.
> 
> What about frontend files? JS, JSX and so on? Should we add those as well

They don't seem to have support for js or ts https://github.com/git/git/blob/master/userdiff.c#L78  as mentioned in this thread https://gist.github.com/tekin/12500956bd56784728e490d8cef9cb81

Comment (Rafi993) [reply .gitattributes:10]: > there's a missing newline at the end of the file

I'll add the newline now.

**Review by rhymes — APPROVED** [review]

File: .gitattributes (hunk 2 above)
Comment (rhymes) [reply .gitattributes]: > They don't seem to have support for js or ts https://github.com/git/git/blob/master/userdiff.c#L78 as mentioned in this thread https://gist.github.com/tekin/12500956bd56784728e490d8cef9cb81

got it, thanks!

**Comment by Rafi993** [comment]
Thank you @rhymes and @citizen428

**Pull request merged**